django_lightweight_queue/worker.py

```diff
@@ -155,12 +159,22 @@ def process(self, backend):
         return True
 
     def tell_master(self, timeout, sigkill_on_stop):
-        self.back_channel.put((
-            self.queue,
-            self.worker_num,
-            timeout,
-            sigkill_on_stop,
-        ))
+        if sigkill_on_stop:
+            # SIGUSR2 can be taken to just cause the process to die
+            # immediately.
+            signal.signal(signal.SIGUSR2, signal.SIG_DFL)
+        else:
+            # SIGUSR2 indicates we should shut down after handling the
+            # next entry.
+            signal.signal(signal.SIGUSR2, self._handle_sigusr2)
+
+        if timeout is not None:
+            # alarm(3) takes whole seconds
+            alarm_duration = int(math.ceil(timeout))
+            signal.alarm(alarm_duration)
+        else:
+            # Cancel any scheduled alarms
+            signal.alarm(0)
```

Comment:
Not sure exactly when `tell_master` is called, but don't we want to also remove the alarm when job processing finishes, otherwise if we had to wait longer than $timeout for the next job, we'd be killed?

Reply:
Similarly, do we want to change the `SIGUSR2` handling for the time between jobs?

Reply:
We do call this after each job, this is behaviour-preserving from the current system.